homeassistant/components/climaveneta_imxw/climate.py

```diff
@@ -0,0 +1,201 @@
+"""Support for the Mitsubishi-Climaveneta iMXW fancoil series."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    SWING_OFF,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.components.modbus import ModbusHub
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ClimavenetaIMXWCoordinator
+from .const import (
+    DOMAIN,
+    MODE_OFF,
+    MODE_ON,
+    MODE_SUMMER,
+    STATE_WRITE_FAN_SPEED_REGISTER,
+    STATE_WRITE_MODE_REGISTER,
+    STATE_WRITE_ON_OFF_REGISTER,
+    TARGET_TEMPERATURE_SUMMER_REGISTER,
+    TARGET_TEMPERATURE_WINTER_REGISTER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CALL_TYPE_WRITE_REGISTER = "write_register"
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the ClimavenetaIMXW Platform.
+
+    Empty as this was the old way of adding platforms.
+    """
+
+
+class ClimavenetaIMXWClimate(
+    CoordinatorEntity[ClimavenetaIMXWCoordinator], ClimateEntity
+):
+    """Representation of a ClimavenetaIMXW fancoil unit."""
+
+    _attr_has_entity_name = True
+    _attr_fan_modes = [FAN_AUTO, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+    _attr_fan_mode = FAN_AUTO
+
+    _attr_hvac_mode = HVACMode.OFF
+    _attr_hvac_modes = [
+        HVACMode.COOL,
+        HVACMode.HEAT,
+        HVACMode.FAN_ONLY,
+        HVACMode.OFF,
+    ]
+
+    _attr_swing_modes = [SWING_OFF]
+    _attr_swing_mode = SWING_OFF
+
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+
+    def __init__(
+        self, coordinator, hub: ModbusHub, modbus_slave: int | None, name: str | None
+    ) -> None:
+        """Initialize the unit."""
+        super().__init__(coordinator)
+        self._hub = hub
+        self._attr_name = name
```

Comment:
We want to avoid users setting custom names in config flows. Instead, please use a device_info object. Users can then rename the device_info. `_attr_name` should be set to `None`.